sd_mining_core/utils/file_utils.py

```diff
@@ -27,61 +56,141 @@ def download_file(base_dir, file_url, file_name):
-            ) and (not config.exclude_sdxl or not model['type'].startswith('sdxl'))
-        }
+        # Fetch configurations
+        models = requests.get(config.model_config_url, timeout=30).json()
+        vaes = requests.get(config.vae_config_url, timeout=30).json()
+        loras = requests.get(config.lora_config_url, timeout=30).json()
 
-        config.vae_configs = {
-            vae['name']: vae
-            for vae in vaes
-        }
+        # If a specific model_id is provided, filter the configurations
+        if config.specified_model_id:
+            specified_model = next((model for model in models if model['name'] == config.specified_model_id), None)
+            if specified_model:
+                config.model_configs = {config.specified_model_id: specified_model}
+                
+                # If it's a composite model, include its base model as well
+                if specified_model.get('type') in ['composite15', 'compositexl']:
+                    base_model_id = specified_model.get('base')
+                    base_model = next((model for model in models if model['name'] == base_model_id), None)
+                    if base_model:
+                        config.model_configs[base_model_id] = base_model
+                    else:
+                        logging.warning(f"Base model '{base_model_id}' for composite model '{config.specified_model_id}' not found.")
+                
+                # Include the corresponding LoRA if it exists
+                lora = next((lora for lora in loras if lora['name'] == config.specified_model_id), None)
+                if lora:
+                    config.lora_configs = {config.specified_model_id: lora}
+            else:
+                # Check if it's a LoRA
+                lora = next((lora for lora in loras if lora['name'] == config.specified_model_id), None)
+                if lora:
+                    config.lora_configs = {config.specified_model_id: lora}
+                    # Include the base model for this LoRA
+                    base_model_id = lora.get('base')
+                    base_model = next((model for model in models if model['name'] == base_model_id), None)
+                    if base_model:
+                        config.model_configs = {base_model_id: base_model}
+                    else:
+                        logging.warning(f"Base model '{base_model_id}' for LoRA '{config.specified_model_id}' not found.")
+                else:
+                    raise ValueError(f"Specified model ID '{config.specified_model_id}' not found in model or LoRA configurations.")
+        else:
+            # Original logic for handling all models
+            config.model_configs = {
+                model['name']: model for model in models
+                if 'type' in model and (
+                    'sd' in model['type'] or 
+                    model['type'].startswith('composite') or
+                    model['type'] == 'flux-dev-4bit'
```

Comment:
Let's remove flux models from the download list. By default, miners don't need to download flux